plugins/inputs/cloudwatch/cloudwatch.go

```diff
@@ -239,168 +412,208 @@ func (c *CloudWatch) updateWindow(relativeTo time.Time) error {
+func (c *CloudWatch) gatherMetrics(
+	params *cloudwatch.GetMetricDataInput,
+) ([]*cloudwatch.MetricDataResult, error) {
+	results := []*cloudwatch.MetricDataResult{}
 
-		resp, err := c.client.ListMetrics(params)
+	for {
+		resp, err := c.client.GetMetricData(params)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to get metric data: %v", err)
 		}
 
-		metrics = append(metrics, resp.Metrics...)
-
-		token = resp.NextToken
-		more = token != nil
-	}
-
-	c.metricCache = &MetricCache{
-		Metrics: metrics,
-		Fetched: time.Now(),
-		TTL:     c.CacheTTL.Duration,
+		results = append(results, resp.MetricDataResults...)
+		if resp.NextToken == nil {
+			break
+		}
+		params.NextToken = resp.NextToken
 	}
 
-	return metrics, nil
+	return results, nil
 }
 
-/*
- * Gather given Metric and emit any error
- */
-func (c *CloudWatch) gatherMetric(
+func (c *CloudWatch) aggregateMetrics(
 	acc telegraf.Accumulator,
-	metric *cloudwatch.Metric,
+	metricDataResults []*cloudwatch.MetricDataResult,
 ) error {
-	params := c.getStatisticsInput(metric)
-	resp, err := c.client.GetMetricStatistics(params)
-	if err != nil {
-		return err
-	}
+	var (
+		grouper   = metric.NewSeriesGrouper()
+		namespace = sanitizeMeasurement(c.Namespace)
+	)
 
-	for _, point := range resp.Datapoints {
+	for _, result := range metricDataResults {
 		tags := map[string]string{
 			"region": c.Region,
-			"unit":   snakeCase(*point.Unit),
 		}
 
-		for _, d := range metric.Dimensions {
-			tags[snakeCase(*d.Name)] = *d.Value
+		if dimensions, ok := c.queryDimensions[*result.Id]; ok {
+			for _, dimension := range *dimensions {
+				tags[dimension.name] = dimension.value
+			}
 		}
 
-		// record field for each statistic
-		fields := map[string]interface{}{}
-
-		if point.Average != nil {
-			fields[formatField(*metric.MetricName, cloudwatch.StatisticAverage)] = *point.Average
-		}
-		if point.Maximum != nil {
-			fields[formatField(*metric.MetricName, cloudwatch.StatisticMaximum)] = *point.Maximum
-		}
-		if point.Minimum != nil {
-			fields[formatField(*metric.MetricName, cloudwatch.StatisticMinimum)] = *point.Minimum
-		}
-		if point.SampleCount != nil {
-			fields[formatField(*metric.MetricName, cloudwatch.StatisticSampleCount)] = *point.SampleCount
-		}
-		if point.Sum != nil {
-			fields[formatField(*metric.MetricName, cloudwatch.StatisticSum)] = *point.Sum
+		for i := range result.Values {
+			grouper.Add(namespace, tags, *result.Timestamps[i], *result.Label, *result.Values[i])
 		}
+	}
 
-		acc.AddFields(formatMeasurement(c.Namespace), fields, tags, *point.Timestamp)
+	for _, metric := range grouper.Metrics() {
+		acc.AddMetric(metric)
 	}
 
 	return nil
 }
 
-/*
- * Formatting helpers
- */
-func formatField(metricName string, statistic string) string {
-	return fmt.Sprintf("%s_%s", snakeCase(metricName), snakeCase(statistic))
+func init() {
+	inputs.Add("cloudwatch", func() telegraf.Input {
+		ttl, _ := time.ParseDuration("1hr")
```

Comment:
"1h" is the right string, but actually since this is a "constant" just do `ttl := time.Hour`, probably without variable assignment